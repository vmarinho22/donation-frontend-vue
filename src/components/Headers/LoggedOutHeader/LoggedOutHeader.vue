<template>
  <v-toolbar density="compact" title="Donation System" color="transparent" style="margin-bottom: 16px;">
    <toggle-theme />
    <v-btn prepend-icon="mdi-account-key" variant="text" @click="goToLogin"> {{ $t('buttons.signin') }} </v-btn>
  </v-toolbar>
</template>

<script setup lang="ts">
import ToggleTheme from '@/components/Buttons/ToggleTheme/ToggleTheme.vue'
import { useRouter } from 'vue-router'

const router = useRouter()

function goToLogin() {
  router.push('/login')
}
</script>
