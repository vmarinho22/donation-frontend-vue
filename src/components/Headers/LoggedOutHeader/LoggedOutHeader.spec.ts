import { describe, it, expect } from 'vitest'

import { mount } from '@vue/test-utils'
import LoggedOutHeader from './LoggedOutHeader.vue'
import vuetify from '@/tests/plugins/vuetify'

describe('ToggleTheme', () => {
  it('renders the component', () => {
    const wrapper = mount(LoggedOutHeader, {
      global: {
        plugins: [vuetify]
      }
    })
    expect(wrapper.html()).toMatchSnapshot();
  })
})