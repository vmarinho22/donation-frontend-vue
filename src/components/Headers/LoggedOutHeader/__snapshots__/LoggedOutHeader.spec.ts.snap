// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`LoggedOutHeader > renders the component 1`] = `
"<header class=\\"v-toolbar v-toolbar--density-compact bg-transparent v-theme--light v-locale--is-ltr\\" style=\\"margin-bottom: 16px;\\">
  <!---->
  <div class=\\"v-toolbar__content\\" style=\\"height: 48px;\\">
    <!---->
    <div class=\\"v-toolbar-title\\">
      <div class=\\"v-toolbar-title__placeholder\\">Donation System
        <!---->
      </div>
    </div><button type=\\"button\\" class=\\"v-btn v-btn--icon v-theme--light v-btn--density-default v-btn--size-default v-btn--variant-text\\"><span class=\\"v-btn__overlay\\"></span><span class=\\"v-btn__underlay\\"></span>
      <!----><span class=\\"v-btn__content\\" data-no-activator=\\"\\"><i class=\\"mdi-weather-night mdi v-icon notranslate v-theme--light v-icon--size-default\\" aria-hidden=\\"true\\"></i></span>
      <!---->
      <!---->
    </button><button type=\\"button\\" class=\\"v-btn v-theme--light v-btn--density-default v-btn--size-default v-btn--variant-text\\"><span class=\\"v-btn__overlay\\"></span><span class=\\"v-btn__underlay\\"></span><span class=\\"v-btn__prepend\\"><i class=\\"mdi-account-key mdi v-icon notranslate v-theme--light v-icon--size-default\\" aria-hidden=\\"true\\"></i></span><span class=\\"v-btn__content\\" data-no-activator=\\"\\">buttons.signin</span>
      <!---->
      <!---->
    </button>
    <!---->
  </div>
  <transition-stub name=\\"expand-transition\\" appear=\\"false\\" persisted=\\"false\\" css=\\"true\\">
    <!---->
  </transition-stub>
</header>"
`;
